### PR TITLE
Migrate to Yahoo Oauth 2.0 API

### PR DESCRIPTION
**Pull request by varbark**
Since OpenID2 will be EOLed on 6/30/20,  
We need to migrate from OpenID2 to OpenID Connect API
https://developer.yahoo.com/oauth2/guide/OpenID2/

SocDir (Social Directory Web Service) also EOL date is June 30,  #2020
https://developer.yahoo.com/oauth/social-directory-eol/

### CODE REVIEW

**Comment by acidjazz** [comment]
@lucasmichot @faustbrian @antonkomarev @Nazariy any idea on when this can happen? we are not passed the EOL date

**Review by acidjazz — APPROVED** [review]

File: Provider.php [hunk 1]
```diff
@@ -57,11 +60,11 @@ protected function getUserByToken($token)
     protected function mapUserToObject(array $user)
     {
         return (new User())->setRaw($user)->map([
-            'id'       => $user['guid'],
+            'id'       => $user['sub'],
             'nickname' => $user['nickname'],
```
Comment (acidjazz) [inline Provider.php:64]: nickname was not present when I tried this out, I'd recommend:

```php
'nickname' => $user['nickname'] ?? null,
```

**Review by acidjazz — APPROVED** [review]
Tested and works, LGTM (minus nickname)

File: Provider.php (hunk 1 above)
Comment (varbark) [reply Provider.php:64]: what will the nickname  value be when nickname was not present ? 
We can change the code as you recommend

Comment (varbark) [reply Provider.php:64]: Also,  if $user['nickname'] not present and the value already is null, 
then  $user['nickname'] ?? null  will just equal to $user['nickname']?

**Comment by lucasmichot** [comment]
@acidjazz - I am not maintainer of this repository

**Review by lucasmichot — CHANGES_REQUESTED** [review]
Please edit this file instead: https://github.com/SocialiteProviders/Providers/blob/master/src/Yahoo/Provider.php

https://github.com/SocialiteProviders/Yahoo is just a split repository of https://github.com/SocialiteProviders/Providers

**Comment by faustbrian** [comment]
@m1guelpf

**Comment by varbark** [comment]
@lucasmichot   thanks your information, I will send PR to the original repository

**Pull request closed without merging**